src/main/java/eu/europeana/metis/sandbox/batch/repository/ExecutionRecordErrorRepository.java

```diff
@@ -21,7 +23,16 @@ public interface ExecutionRecordErrorRepository extends
    * @param executionName The name of the executionRun.
    * @return A list of ExecutionRecordException entities matching the specified dataset ID and executionRun name.
    */
-  List<ExecutionRecordError> findByExecutionRun_DatasetIdAndExecutionRun_ExecutionName(String datasetId, String executionName);
+  @Query("""
+      SELECT e.identifier AS identifier,
+             e.exception AS exception
+      FROM ExecutionRecordError e
+      WHERE e.executionRun.datasetId = :datasetId
+        AND e.executionRun.executionName = :executionName
+      """)
+  Stream<ExecutionRecordErrorProjection> findErrorsWithIdentifiers(
```

Comment:
regarding the new style of fetching rows I would suggest also adding this to complement the amount of records while streaming in all the places where `Stream` is returned.
```
@QueryHints(@QueryHint(name = AvailableHints.HINT_FETCH_SIZE, value = "25"))
```